### DISCUSSION TITLE
Datagrid editing state

### DISCUSSION THREAD

**Discussion by davidjhurtado** [Megabit/Blazorise#3098] [Q&A] [answered]
We been deploy an app for register information, but on many cases users star to add or edit a detail but they didn't press the save button on datagrid control, they press save button on the main form, in that cases the information registered in the datagrid row is lost, i wonder is there a way to get the state of datagrid?, to send a message to the user, and inform they that exists data incomplete or information to loss.

**Comment by David-Moreira** [comment, marked as answer]
Hello,
Override the new / edit Datagrid buttons so that you set an InEdit flag to true. It is then set to false on the Datagrids provided save handlers... 

If your user presses Save on Main form and InEdit is set to true it can warn them they haven't saved the grid, before closing.

Just let us know if these generic steps helps you or if you really need more detailed information / example.

**Comment by davidjhurtado** [comment]
Thanks for your answer, it's works for us.
